### PR TITLE
Add support for Python 3.8

### LINKED ISSUE #254: Support and deploy for Python 3.8
https://docs.python.org/3/whatsnew/3.8.html

> Python 3.8 was released on October 14th, 2019.

I think it would be important to have a release version that officially supports Python 3.8 and also deploy for it.

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR tests signac on Python 3.8. I am not aware of any other configuration needed to officially support Python 3.8.

## Motivation and Context
Resolves #254.

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac/blob/master/changelog.txt) and added all related issue and pull request numbers for future reference (if applicable). See example below.

Example for a changelog entry: `Fix issue with launching rockets to the moon (#101, #212).`

### CODE REVIEW

**Comment by bdice** [comment]
There is a problem with running `codecov` on Python 3.8. For now, I'm skipping codecov on Python 3.8.

Failed test log: https://circleci.com/gh/glotzerlab/signac/4308
```python
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/runpy.py", line 183, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/usr/local/lib/python3.8/runpy.py", line 142, in _get_module_details
    return _get_module_details(pkg_main_name, error)
  File "/usr/local/lib/python3.8/runpy.py", line 109, in _get_module_details
    __import__(pkg_name)
  File "/home/circleci/repo/venv/lib/python3.8/site-packages/coverage/__init__.py", line 13, in <module>
    from coverage.control import Coverage, process_startup
  File "/home/circleci/repo/venv/lib/python3.8/site-packages/coverage/control.py", line 29, in <module>
    from coverage.python import PythonFileReporter
  File "/home/circleci/repo/venv/lib/python3.8/site-packages/coverage/python.py", line 11, in <module>
    from coverage.parser import PythonParser
  File "/home/circleci/repo/venv/lib/python3.8/site-packages/coverage/parser.py", line 320, in <module>
    OP_BREAK_LOOP = _opcode('BREAK_LOOP')
  File "/home/circleci/repo/venv/lib/python3.8/site-packages/coverage/parser.py", line 278, in _opcode
    return dis.opmap[name]
KeyError: 'BREAK_LOOP'
```

Potentially related issue: https://github.com/nedbat/coveragepy/issues/772

**Comment by b-butler** [comment]
If we plan on merging this before codecov has a fix, then I suggest we document that we need to come back to adding code coverage for python 3.8 in an issue.

**Comment by bdice** [comment]
> If we plan on merging this before codecov has a fix, then I suggest we document that we need to come back to adding code coverage for python 3.8 in an issue.

Agreed - I was documenting it for that reason. #259 has been created.

File: tests/test_project.py [hunk 1]
```diff
@@ -201,7 +201,11 @@ def test_no_workspace_warn_on_find(self):
         self.assertFalse(os.path.exists(self.project.workspace()))
         with self.assertLogs(level='INFO') as cm:
             list(self.project.find_jobs())
-            self.assertEqual(len(cm.output), 2)
+            # Python < 3.8 will return 2 messages.
+            # Python >= 3.8 will return 3 messages, because it determines the
+            # length of the project one additional time during the list
+            # constructor: https://bugs.python.org/issue33234
```
Comment (vyasr) [inline tests/test_project.py]: Thanks for including the comment, that's what I would have requested :)

**Pull request merged**